internal/guest/storage/scsi/scsi.go

```diff
@@ -38,7 +38,7 @@ var (
 // If `encrypted` is set to true, the SCSI device will be encrypted using
 // dm-crypt.
 func Mount(ctx context.Context, controller, lun uint8, target string, readonly bool, encrypted bool, options []string) (err error) {
-	ctx, span := trace.StartSpan(ctx, "scsi::Mount")
+	spnCtx, span := trace.StartSpan(ctx, "scsi::Mount")
```

Comment:
Was there a bug here with reassigning ctx and using it?

Reply:
I just wanted make sure that we're waiting on the right context when we do `ctx.Done()`. But probably span's context would also be notified in that case.

Reply:
Could be, I always get confused when reassigning the contexts for spans. I'd ran into an issue just like what you're describing for ncproxy so I guess better to be safe than sorry